### DISCUSSION TITLE
How to handle different NuGet packages for different target .NET versions

### DISCUSSION THREAD

**Discussion by Piedone** [renovatebot/renovate#33104] [Request Help]
### How are you running Renovate?

A Mend.io-hosted app

### Please tell us more about your question or problem

In [Orchard Core](https://github.com/OrchardCMS/OrchardCore) we just started to use Renovate for everything but NPM package updates. In our [`Directory.Packages.props` file](https://github.com/OrchardCMS/OrchardCore/blob/3fcc0c39378d4ee27378ddfd112c3abd77aa3037/Directory.Packages.props) we double-target .NET 8 and 9, since it's our policy to support the current LTS and current latest .NET versions. You can see how this targeting works for package versions on the lines 123, 130, 137, and 156.

Ideally, we'd like Renovate to honor such targeting, and at least keep the targets separate: e.g., if we ignore an update (close a Renovate PR) for a package for one target, the other target should be considered different.

See e.g. [this PR](https://github.com/OrchardCMS/OrchardCore/pull/17241/files) where Renovate tried to update the .NET 8 Serilog version, which failed (this is why we didn't do this update). For .NET 9, we deliberately use a different, more recent 9.x version (line 153). Closing this PR shouldn't put the .NET 9 package reference on ignore, but only the .NET 8 one.

Is this possible?

Ideally, we'd like Renovate to suggest only compatible updates in any case, but this was tricky so I wouldn't expect Renovate to know:

```
/home/runner/work/OrchardCore/OrchardCore/src/OrchardCore/OrchardCore.Logging.Serilog/OrchardCore.Logging.Serilog.csproj : error NU1109: Detected package downgrade: System.Text.Json from 9.0.0 to centrally defined 8.0.5. Update the centrally managed package version to a higher version.  [/home/runner/work/OrchardCore/OrchardCore/OrchardCore.sln]
/home/runner/work/OrchardCore/OrchardCore/src/OrchardCore/OrchardCore.Logging.Serilog/OrchardCore.Logging.Serilog.csproj : error NU1109:  OrchardCore.Logging.Serilog -> Serilog.AspNetCore 9.0.0 -> Serilog.Settings.Configuration 9.0.0 -> Microsoft.Extensions.DependencyModel 9.0.0 -> System.Text.Json (>= 9.0.0)  [/home/runner/work/OrchardCore/OrchardCore/OrchardCore.sln]
/home/runner/work/OrchardCore/OrchardCore/src/OrchardCore/OrchardCore.Logging.Serilog/OrchardCore.Logging.Serilog.csproj : error NU1109:  OrchardCore.Logging.Serilog -> System.Text.Json (>= 8.0.5) [/home/runner/work/OrchardCore/OrchardCore/OrchardCore.sln]
```

### Logs (if relevant)

<details><summary>Logs</summary>

```

Replace this text with your logs, between the starting and ending triple backticks

```

</details>

**Comment by RahulGautamSingh** [comment]
Check `allowedVersions` [^1]
Since, you manage the LTS & latest versions, I hope it's not much trouble to update the allowedVersions each time you change LTS and latest.

[^1]: https://docs.renovatebot.com/configuration-options/#allowedversions

**Reply by Piedone** [reply]
Thank you! How would that work together with .NET version matching, though? I can't find any way to do that in a Renovate config file (i.e. something along the lines of "this package rule is only active if the project is built for .NET 8").

The closest I could come up, what works in this case but is not a generic solution, is matching on the `Serilog.AspNetCore` version (assuming that its 8.x versions will always only target .NET 8, and anything >v8 will be .NET 9 for now):

```
packageRules: [
    {
        // The .NET 8 version of Serilog.AspNetCore (8.x) needs to stay on 8.x. We maintain a separate reference to
        // the 9.x version in Directory.Packages.props, only active when building for .NET 9.
        matchPackageNames: [
            'Serilog.AspNetCore',
        ],
        allowedVersions: '<9.0.0',
        matchCurrentVersion: '<9.0.0',
    },
]
```

**Reply by RahulGautamSingh** [reply]
So, you can try package rules, use the matcher packageName to target all deps which you want to target then narrow it to the LTS version by using matchCurrentVersion and use allowedVersions to restrict it to <9.

```json5
packageRules: [
    {
        matchPackageNames: [
            "Microsoft.Extensions**",
            "Microsoft.AspNetCore**",
            "System.Text.Json",
            "Serilog.AspNetCore"
        ],
        allowedVersions: '<9.0.0',
        matchCurrentVersion: '<9.0.0',
    },
]
```

You will need to update this package rule whenever you update the LTS version though.

**Reply by Piedone** [reply]
Thank you! I tried this, but it didn't seem to work, Renovate opened this PR: https://github.com/OrchardCMS/OrchardCore/pull/17259 I configured almost what you have above: https://github.com/OrchardCMS/OrchardCore/pull/17249 What do you think might be the issue?

**Comment by CNamratha18** [comment]
I want to target specifically only this version sspc_PR6-SNAPSHOT but this does not work only it targets

sspc_.*|^[0-9]+(\\.[0-9]+)*)$/ and based on this updates are happening
 "packageRules": [
    {
      "groupName": "all dependencies",
      "matchDatasources": [
        "artifactory"
      ],
      "allowedVersions": "/^(sspc_PR6-SNAPSHOT|sspc_.*|^[0-9]+(\\.[0-9]+)*)$/",
      "versioning": "auto"
    }
  ]                                  if iam trying this  it does not work                                                                                                              
                   
"packageRules": [
  {
    "groupName": "all dependencies",
    "matchDatasources": ["artifactory"],
    "matchPackageNames": ["ssp/component/GlobalCoreApp"],
    "allowedVersions": "/^sspc_PR6-SNAPSHOT$/",
    "versioning": "semver"
  }
]

**Reply by CNamratha18** [reply]
output is this one DEBUG: Adding password authentication for git-ssp.cicd.rnd.amadeus.net (hostType=bitbucket-server) to hostRules
 INFO: Repository started (repository=sspc/testkafka)
       "renovateVersion": "37.440.6"
DEBUG: Using localDir: C:/Users/NCHAKR~1/AppData/Local/Temp/renovate/repos/bitbucket-server/sspc/testkafka (repository=sspc/testkafka)
DEBUG: PackageFiles.clear() - Package files deleted (repository=sspc/testkafka)
DEBUG: initRepo("{
         "repository": "sspc/testkafka"
       }") (repository=sspc/testkafka)
DEBUG: hostRules: applying Basic authentication for git-ssp.cicd.rnd.amadeus.net (repository=sspc/testkafka)
DEBUG: Using queue: host=git-ssp.cicd.rnd.amadeus.net, concurrency=16 (repository=sspc/testkafka)
DEBUG: sspc/testkafka owner = SSPC (repository=sspc/testkafka)
DEBUG: Resetting npmrc (repository=sspc/testkafka)
DEBUG: Resetting npmrc (repository=sspc/testkafka)
DEBUG: checkOnboarding() (repository=sspc/testkafka)
DEBUG: isOnboarded() (repository=sspc/testkafka)
DEBUG: findPr(renovate/configure, "Configure Renovate", "!open") (repository=sspc/testkafka)
DEBUG: getPrList() (repository=sspc/testkafka)
DEBUG: Retrieved Pull Requests, count: 20 (repository=sspc/testkafka)
DEBUG: Renovate did not find a PR for branch #renovate/configure (repository=sspc/testkafka)
DEBUG: findPr(renovate/configure, "chore: Configure Renovate", "!open") (repository=sspc/testkafka)
DEBUG: getPrList() (repository=sspc/testkafka)
DEBUG: returning cached PR list (repository=sspc/testkafka)
DEBUG: Renovate did not find a PR for branch #renovate/configure (repository=sspc/testkafka)
DEBUG: findFile(renovate.json) (repository=sspc/testkafka)
DEBUG: Initializing git repository into C:/Users/NCHAKR~1/AppData/Local/Temp/renovate/repos/bitbucket-server/sspc/testkafka (repository=sspc/testkafka)
DEBUG: Performing blobless clone (repository=sspc/testkafka)
DEBUG: git clone completed (repository=sspc/testkafka)
       "durationMs": 2725
DEBUG: latest repository commit (repository=sspc/testkafka)
       "latestCommit": {
         "hash": "4da1cf230782322bf0da6e920f2f3a27c2478747",
         "date": "2025-01-08T06:24:58Z",
         "message": "build.yaml edited online with Bitbucket",
         "refs": "HEAD -> develop, origin/develop, origin/HEAD",
         "body": "",
         "author_name": "Namratha CHAKRAVARTHULA (contractor)",
         "author_email": "namratha.chakravarthula@amadeus.com"
       }
DEBUG: Config file exists, fileName: renovate.json (repository=sspc/testkafka)
DEBUG: ensureIssueClosing(Action required: Add a Renovate config) (repository=sspc/testkafka)
DEBUG: Repo is onboarded (repository=sspc/testkafka)
DEBUG: Found renovate.json config file (repository=sspc/testkafka)
DEBUG: Repository config (repository=sspc/testkafka)
       "fileName": "renovate.json",
       "config": {
         "$schema": "https://docs.renovatebot.com/renovate-schema.json",
         "description": "SSP Package Envelop version update",
         "extends": ["local>SSPC/renovate-bot//config/default"]
       }
DEBUG: migrateAndValidate() (repository=sspc/testkafka)
DEBUG: No config migration necessary (repository=sspc/testkafka)
DEBUG: Post-massage config (repository=sspc/testkafka)
       "config": {
         "$schema": "https://docs.renovatebot.com/renovate-schema.json",
         "description": ["SSP Package Envelop version update"],
         "extends": ["config:recommended", "local>SSPC/renovate-bot//config/default"]
       }
 WARN: File contents are invalid JSON but parse using JSON5. Support for this will be removed in a future release so please change to a support .json5 file name or ensure correct JSON syntax. (repository=sspc/testkafka)
       "context": "config/base.json"
DEBUG: Setting hostRules from config (repository=sspc/testkafka)
DEBUG: Found repo ignorePaths (repository=sspc/testkafka)
       "ignorePaths": [
         "**/node_modules/**",
         "**/bower_components/**",
         "**/vendor/**",
         "**/examples/**",
         "**/__tests__/**",
         "**/test/**",
         "**/tests/**",
         "**/__fixtures__/**"
       ]
DEBUG: No vulnerability alerts found (repository=sspc/testkafka)
DEBUG: baseBranches (repository=sspc/testkafka)
       "baseBranches": ["develop", "main"]
DEBUG: baseBranch: develop (repository=sspc/testkafka, baseBranch=develop)
DEBUG: extract() (repository=sspc/testkafka, baseBranch=develop)
DEBUG: Setting current branch to develop (repository=sspc/testkafka, baseBranch=develop)
DEBUG: latest commit (repository=sspc/testkafka, baseBranch=develop)
       "branchName": "develop",
       "latestCommitDate": "2025-01-08T06:24:58Z"
DEBUG: Using file match: (^|/)tasks/[^/]+\.ya?ml$ for manager ansible (repository=sspc/testkafka, baseBranch=develop)
DEBUG: Using file match: (^|/)(galaxy|requirements)(\.ansible)?\.ya?ml$ for manager ansible-galaxy (repository=sspc/testkafka, baseBranch=develop)
DEBUG: Using file match: (^|/)\.tool-versions$ for manager asdf (repository=sspc/testkafka, baseBranch=develop)
DEBUG: Using file match: azure.*pipelines?.*\.ya?ml$ for manager azure-pipelines (repository=sspc/testkafka, baseBranch=develop)
DEBUG: Using file match: (^|/)batect(-bundle)?\.ya?ml$ for manager batect (repository=sspc/testkafka, baseBranch=develop)
DEBUG: Using file match: (^|/)batect$ for manager batect-wrapper (repository=sspc/testkafka, baseBranch=develop)
DEBUG: Using file match: (^|/)WORKSPACE(|\.bazel|\.bzlmod)$ for manager bazel (repository=sspc/testkafka, baseBranch=develop)
DEBUG: Using file match: \.bzl$ for manager bazel (repository=sspc/testkafka, baseBranch=develop)
DEBUG: Using file match: (^|/)MODULE\.bazel$ for manager bazel-module (repository=sspc/testkafka, baseBranch=develop)
DEBUG: Using file match: (^|/)\.bazelversion$ for manager bazelisk (repository=sspc/testkafka, baseBranch=develop)
DEBUG: Using file match: \.bicep$ for manager bicep (repository=sspc/testkafka, baseBranch=develop)
DEBUG: Using file match: (^|/)\.?bitbucket-pipelines\.ya?ml$ for manager bitbucket-pipelines (repository=sspc/testkafka, baseBranch=develop)
DEBUG: Using file match: buildkite\.ya?ml for manager buildkite (repository=sspc/testkafka, baseBranch=develop)
DEBUG: Using file match: \.buildkite/.+\.ya?ml$ for manager buildkite (repository=sspc/testkafka, baseBranch=develop)
DEBUG: Using file match: (^|/)bun\.lockb$ for manager bun (repository=sspc/testkafka, baseBranch=develop)
DEBUG: Using file match: (^|/)Gemfile$ for manager bundler (repository=sspc/testkafka, baseBranch=develop)
DEBUG: Using file match: \.cake$ for manager cake (repository=sspc/testkafka, baseBranch=develop)
DEBUG: Using file match: (^|/)Cargo\.toml$ for manager cargo (repository=sspc/testkafka, baseBranch=develop)
DEBUG: Using file match: (^|/)\.circleci/.+\.ya?ml$ for manager circleci (repository=sspc/testkafka, baseBranch=develop)
DEBUG: Using file match: (^|/)cloudbuild\.ya?ml for manager cloudbuild (repository=sspc/testkafka, baseBranch=develop)
DEBUG: Using file match: (^|/)Podfile$ for manager cocoapods (repository=sspc/testkafka, baseBranch=develop)
DEBUG: Using file match: (^|/)([\w-]*)composer\.json$ for manager composer (repository=sspc/testkafka, baseBranch=develop)
DEBUG: Using file match: (^|/)conanfile\.(txt|py)$ for manager conan (repository=sspc/testkafka, baseBranch=develop)
DEBUG: Using file match: (^|/)cpanfile$ for manager cpanfile (repository=sspc/testkafka, baseBranch=develop)
DEBUG: Using file match: (^|/)(?:deps|bb)\.edn$ for manager deps-edn (repository=sspc/testkafka, baseBranch=develop)
DEBUG: Using file match: ^.devcontainer/devcontainer.json$ for manager devcontainer (repository=sspc/testkafka, baseBranch=develop)
DEBUG: Using file match: ^.devcontainer.json$ for manager devcontainer (repository=sspc/testkafka, baseBranch=develop)
DEBUG: Using file match: (^|/)(?:docker-)?compose[^/]*\.ya?ml$ for manager docker-compose (repository=sspc/testkafka, baseBranch=develop)
DEBUG: Using file match: (^|/|\.)([Dd]ocker|[Cc]ontainer)file$ for manager dockerfile (repository=sspc/testkafka, baseBranch=develop)
DEBUG: Using file match: (^|/)([Dd]ocker|[Cc]ontainer)file[^/]*$ for manager dockerfile (repository=sspc/testkafka, baseBranch=develop)
DEBUG: Using file match: (^|/)\.drone\.yml$ for manager droneci (repository=sspc/testkafka, baseBranch=develop)
DEBUG: Using file match: (^|/)fleet\.ya?ml for manager fleet (repository=sspc/testkafka, baseBranch=develop)
DEBUG: Using file match: (?:^|/)gotk-components\.ya?ml$ for manager flux (repository=sspc/testkafka, baseBranch=develop)
DEBUG: Using file match: (^|/)\.fvm/fvm_config\.json$ for manager fvm (repository=sspc/testkafka, baseBranch=develop)
DEBUG: Using file match: (^|/)\.fvmrc$ for manager fvm (repository=sspc/testkafka, baseBranch=develop)
DEBUG: Using file match: (^|/)\.gitmodules$ for manager git-submodules (repository=sspc/testkafka, baseBranch=develop)
DEBUG: Using file match: (^|/)(workflow-templates|\.(?:github|gitea|forgejo)/(?:workflows|actions))/.+\.ya?ml$ for manager github-actions (repository=sspc/testkafka, baseBranch=develop)
DEBUG: Using file match: (^|/)action\.ya?ml$ for manager github-actions (repository=sspc/testkafka, baseBranch=develop)
DEBUG: Using file match: \.gitlab-ci\.ya?ml$ for manager gitlabci (repository=sspc/testkafka, baseBranch=develop)
DEBUG: Using file match: \.gitlab-ci\.ya?ml$ for manager gitlabci-include (repository=sspc/testkafka, baseBranch=develop)
DEBUG: Using file match: (^|/)go\.mod$ for manager gomod (repository=sspc/testkafka, baseBranch=develop)
DEBUG: Using file match: \.gradle(\.kts)?$ for manager gradle (repository=sspc/testkafka, baseBranch=develop)
DEBUG: Using file match: (^|/)gradle\.properties$ for manager gradle (repository=sspc/testkafka, baseBranch=develop)
DEBUG: Using file match: (^|/)gradle/.+\.toml$ for manager gradle (repository=sspc/testkafka, baseBranch=develop)
DEBUG: Using file match: (^|/)buildSrc/.+\.kt$ for manager gradle (repository=sspc/testkafka, baseBranch=develop)
DEBUG: Using file match: \.versions\.toml$ for manager gradle (repository=sspc/testkafka, baseBranch=develop)
DEBUG: Using file match: (^|/)versions.props$ for manager gradle (repository=sspc/testkafka, baseBranch=develop)
DEBUG: Using file match: (^|/)versions.lock$ for manager gradle (repository=sspc/testkafka, baseBranch=develop)
DEBUG: Using file match: (^|/)gradle/wrapper/gradle-wrapper\.properties$ for manager gradle-wrapper (repository=sspc/testkafka, baseBranch=develop)
DEBUG: Using file match: (^|/)requirements\.ya?ml$ for manager helm-requirements (repository=sspc/testkafka, baseBranch=develop)
DEBUG: Using file match: (^|/)values\.ya?ml$ for manager helm-values (repository=sspc/testkafka, baseBranch=develop)
DEBUG: Using file match: (^|/)helmfile\.ya?ml(?:\.gotmpl)?$ for manager helmfile (repository=sspc/testkafka, baseBranch=develop)
DEBUG: Using file match: (^|/)Chart\.ya?ml$ for manager helmv3 (repository=sspc/testkafka, baseBranch=develop)
DEBUG: Using file match: (^|/)bin/hermit$ for manager hermit (repository=sspc/testkafka, baseBranch=develop)
DEBUG: Using file match: ^Formula/[^/]+[.]rb$ for manager homebrew (repository=sspc/testkafka, baseBranch=develop)
DEBUG: Using file match: \.html?$ for manager html (repository=sspc/testkafka, baseBranch=develop)
DEBUG: Using file match: (^|/)plugins\.(txt|ya?ml)$ for manager jenkins (repository=sspc/testkafka, baseBranch=develop)
DEBUG: Using file match: (^|/)jsonnetfile\.json$ for manager jsonnet-bundler (repository=sspc/testkafka, baseBranch=develop)
DEBUG: Using file match: ^.+\.main\.kts$ for manager kotlin-script (repository=sspc/testkafka, baseBranch=develop)
DEBUG: Using file match: (^|/)kustomization\.ya?ml$ for manager kustomize (repository=sspc/testkafka, baseBranch=develop)
DEBUG: Using file match: (^|/)project\.clj$ for manager leiningen (repository=sspc/testkafka, baseBranch=develop)
DEBUG: Using file match: (^|/|\.)pom\.xml$ for manager maven (repository=sspc/testkafka, baseBranch=develop)
DEBUG: Using file match: ^(((\.mvn)|(\.m2))/)?settings\.xml$ for manager maven (repository=sspc/testkafka, baseBranch=develop)
DEBUG: Using file match: (^|/)\.mvn/extensions\.xml$ for manager maven (repository=sspc/testkafka, baseBranch=develop)
DEBUG: Using file match: (^|\/).mvn/wrapper/maven-wrapper.properties$ for manager maven-wrapper (repository=sspc/testkafka, baseBranch=develop)
DEBUG: Using file match: (^|/)package\.js$ for manager meteor (repository=sspc/testkafka, baseBranch=develop)
DEBUG: Using file match: (^|/)Mintfile$ for manager mint (repository=sspc/testkafka, baseBranch=develop)
DEBUG: Using file match: (^|/)\.?mise\.toml$ for manager mise (repository=sspc/testkafka, baseBranch=develop)
DEBUG: Using file match: (^|/)\.?mise/config\.toml$ for manager mise (repository=sspc/testkafka, baseBranch=develop)
DEBUG: Using file match: (^|/)mix\.exs$ for manager mix (repository=sspc/testkafka, baseBranch=develop)
DEBUG: Using file match: (^|/)flake\.nix$ for manager nix (repository=sspc/testkafka, baseBranch=develop)
DEBUG: Using file match: (^|/)\.node-version$ for manager nodenv (repository=sspc/testkafka, baseBranch=develop)
DEBUG: Using file match: (^|/)package\.json$ for manager npm (repository=sspc/testkafka, baseBranch=develop)
DEBUG: Using file match: \.(?:cs|fs|vb)proj$ for manager nuget (repository=sspc/testkafka, baseBranch=develop)
DEBUG: Using file match: \.(?:props|targets)$ for manager nuget (repository=sspc/testkafka, baseBranch=develop)
DEBUG: Using file match: (^|/)dotnet-tools\.json$ for manager nuget (repository=sspc/testkafka, baseBranch=develop)
DEBUG: Using file match: (^|/)global\.json$ for manager nuget (repository=sspc/testkafka, baseBranch=develop)
DEBUG: Using file match: (^|/)\.nvmrc$ for manager nvm (repository=sspc/testkafka, baseBranch=develop)
DEBUG: Using file match: (^|/)src/main/features/.+\.json$ for manager osgi (repository=sspc/testkafka, baseBranch=develop)
DEBUG: Using file match: (^|/)pyproject\.toml$ for manager pep621 (repository=sspc/testkafka, baseBranch=develop)
DEBUG: Using file match: (^|/)[\w-]*requirements([-.]\w+)?\.(txt|pip)$ for manager pip_requirements (repository=sspc/testkafka, baseBranch=develop)
DEBUG: Using file match: (^|/)setup\.py$ for manager pip_setup (repository=sspc/testkafka, baseBranch=develop)
DEBUG: Using file match: (^|/)Pipfile$ for manager pipenv (repository=sspc/testkafka, baseBranch=develop)
DEBUG: Using file match: (^|/)pyproject\.toml$ for manager poetry (repository=sspc/testkafka, baseBranch=develop)
DEBUG: Using file match: (^|/)\.pre-commit-config\.ya?ml$ for manager pre-commit (repository=sspc/testkafka, baseBranch=develop)
DEBUG: Using file match: (^|/)pubspec\.ya?ml$ for manager pub (repository=sspc/testkafka, baseBranch=develop)
DEBUG: Using file match: (^|/)Puppetfile$ for manager puppet (repository=sspc/testkafka, baseBranch=develop)
DEBUG: Using file match: (^|/)\.python-version$ for manager pyenv (repository=sspc/testkafka, baseBranch=develop)
DEBUG: Using file match: (^|/)\.ruby-version$ for manager ruby-version (repository=sspc/testkafka, baseBranch=develop)
DEBUG: Using file match: (^|/)runtime.txt$ for manager runtime-version (repository=sspc/testkafka, baseBranch=develop)
DEBUG: Using file match: \.sbt$ for manager sbt (repository=sspc/testkafka, baseBranch=develop)
DEBUG: Using file match: project/[^/]*\.scala$ for manager sbt (repository=sspc/testkafka, baseBranch=develop)
DEBUG: Using file match: project/build\.properties$ for manager sbt (repository=sspc/testkafka, baseBranch=develop)
DEBUG: Using file match: (^|/)repositories$ for manager sbt (repository=sspc/testkafka, baseBranch=develop)
DEBUG: Using file match: (^|/)\.scalafmt.conf$ for manager scalafmt (repository=sspc/testkafka, baseBranch=develop)
DEBUG: Using file match: (^|/)setup\.cfg$ for manager setup-cfg (repository=sspc/testkafka, baseBranch=develop)
DEBUG: Using file match: (^|/)Package\.swift for manager swift (repository=sspc/testkafka, baseBranch=develop)
DEBUG: Using file match: \.tf$ for manager terraform (repository=sspc/testkafka, baseBranch=develop)
DEBUG: Using file match: (^|/)\.terraform-version$ for manager terraform-version (repository=sspc/testkafka, baseBranch=develop)
DEBUG: Using file match: (^|/)terragrunt\.hcl$ for manager terragrunt (repository=sspc/testkafka, baseBranch=develop)
DEBUG: Using file match: (^|/)\.terragrunt-version$ for manager terragrunt-version (repository=sspc/testkafka, baseBranch=develop)
DEBUG: Using file match: \.tflint\.hcl$ for manager tflint-plugin (repository=sspc/testkafka, baseBranch=develop)
DEBUG: Using file match: ^\.travis\.ya?ml$ for manager travis (repository=sspc/testkafka, baseBranch=develop)
DEBUG: Using file match: (^|/)\.vela\.ya?ml$ for manager velaci (repository=sspc/testkafka, baseBranch=develop)
DEBUG: Using file match: (^|/)vendir\.yml$ for manager vendir (repository=sspc/testkafka, baseBranch=develop)
DEBUG: Using file match: ^\.woodpecker(?:/[^/]+)?\.ya?ml$ for manager woodpecker (repository=sspc/testkafka, baseBranch=develop)
DEBUG: Using file match: build\.yaml$ for manager regex (repository=sspc/testkafka, baseBranch=develop)
DEBUG: Using file match: ^.*\.?build\.sh$ for manager regex (repository=sspc/testkafka, baseBranch=develop)
DEBUG: Using file match: ^.*\.?README\.md$ for manager regex (repository=sspc/testkafka, baseBranch=develop)
DEBUG: Matched 1 file(s) for manager regex: build.yaml (repository=sspc/testkafka, baseBranch=develop)
DEBUG: Matched 1 file(s) for manager regex: RDKafka/build.sh (repository=sspc/testkafka, baseBranch=develop)
DEBUG: Matched 1 file(s) for manager regex: README.md (repository=sspc/testkafka, baseBranch=develop)
DEBUG: manager extract durations (ms) (repository=sspc/testkafka, baseBranch=develop)
       "managers": {"regex": 8}
DEBUG: Found regex package files (repository=sspc/testkafka, baseBranch=develop)
DEBUG: Found regex package files (repository=sspc/testkafka, baseBranch=develop)
DEBUG: Found regex package files (repository=sspc/testkafka, baseBranch=develop)
DEBUG: Found 3 package file(s) (repository=sspc/testkafka, baseBranch=develop)
 INFO: Dependency extraction complete (repository=sspc/testkafka, baseBranch=develop)
       "stats": {
         "managers": {"regex": {"fileCount": 3, "depCount": 4}},
         "total": {"fileCount": 3, "depCount": 4}
       }
 WARN: Base branch does not exist - skipping (repository=sspc/testkafka, baseBranch=main)
DEBUG: baseBranch: develop (repository=sspc/testkafka, baseBranch=develop)
DEBUG: Found no satisfying versions with 'semver' versioning (repository=sspc/testkafka, baseBranch=develop)
DEBUG: PackageFiles.add() - Package file saved for base branch (repository=sspc/testkafka, baseBranch=develop)
DEBUG: Package releases lookups complete (repository=sspc/testkafka, baseBranch=develop)
DEBUG: branchifyUpgrades (repository=sspc/testkafka, baseBranch=develop)
DEBUG: detectSemanticCommits() (repository=sspc/testkafka, baseBranch=develop)
DEBUG: getCommitMessages (repository=sspc/testkafka, baseBranch=develop)
DEBUG: semanticCommits: detected "unknown" (repository=sspc/testkafka, baseBranch=develop)
DEBUG: semanticCommits: disabled (repository=sspc/testkafka, baseBranch=develop)
DEBUG: 0 flattened updates found:  (repository=sspc/testkafka, baseBranch=develop)
DEBUG: Returning 0 branch(es) (repository=sspc/testkafka, baseBranch=develop)
DEBUG: config.repoIsOnboarded=true (repository=sspc/testkafka, baseBranch=develop)
DEBUG: packageFiles with updates (repository=sspc/testkafka, baseBranch=develop)
       "config": {
         "regex": [
           {
             "deps": [
               {
                 "depName": "ssp/component/GlobalCoreApp",
                 "currentValue": "3.19.0",
                 "datasource": "artifactory",
                 "registryUrls": ["https://repository.rnd.amadeus.net/ssp-built"],
                 "replaceString": "version: 3.19.0",
                 "updates": [],
                 "packageName": "ssp/component/GlobalCoreApp",
                 "versioning": "semver",
                 "warnings": [],
                 "registryUrl": "https://repository.rnd.amadeus.net/ssp-built",
                 "currentVersion": "3.19.0",
                 "currentVersionTimestamp": "2024-06-24T14:28:00.000Z",
                 "fixedVersion": "3.19.0"
               },
               {
                 "depName": "ssp/component/GlobalCoreApp",
                 "currentValue": "3.20.0",
                 "datasource": "artifactory",
                 "registryUrls": ["https://repository.rnd.amadeus.net/ssp-built"],
                 "replaceString": "version: 3.20.0",
                 "updates": [],
                 "packageName": "ssp/component/GlobalCoreApp",
                 "versioning": "semver",
                 "warnings": [],
                 "registryUrl": "https://repository.rnd.amadeus.net/ssp-built",
                 "currentVersion": "3.20.0",
                 "fixedVersion": "3.20.0"
               }
             ],

**Reply by n0099** [reply]
Could you wrap these logs in codeblock like ```?
